spec/rubocop/cop/style/redundant_line_continuation_spec.rb

```diff
@@ -0,0 +1,196 @@
+                        ^ Redundant line continuation.
+                    baz)
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      def self.foo(bar,#{trailing_whitespace}
+                    baz)
+      end
+    RUBY
+  end
+
+  it 'registers an offense when redundant line continuations for block' do
+    expect_offense(<<~'RUBY')
+      foo do \
+             ^ Redundant line continuation.
+        bar
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo do#{trailing_whitespace}
+        bar
+      end
+    RUBY
+  end
+
+  it 'registers an offense when redundant line continuations for method chain' do
+    expect_offense(<<~'RUBY')
+      foo. \
+           ^ Redundant line continuation.
+        bar
+
+      foo \
+          ^ Redundant line continuation.
+        .bar \
+             ^ Redundant line continuation.
+          .baz
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo.#{trailing_whitespace}
+        bar
+
+      foo#{trailing_whitespace}
+        .bar#{trailing_whitespace}
+          .baz
+    RUBY
+  end
+
+  it 'registers an offense when redundant line continuations for method chain with safe navigation' do
+    expect_offense(<<~'RUBY')
+      foo&. \
+            ^ Redundant line continuation.
+        bar
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo&.#{trailing_whitespace}
+        bar
+    RUBY
+  end
+
+  it 'registers an offense when redundant line continuations for array' do
+    expect_offense(<<~'RUBY')
+      [foo, \
+            ^ Redundant line continuation.
+        bar]
+    RUBY
+
+    expect_correction(<<~RUBY)
+      [foo,#{trailing_whitespace}
+        bar]
+    RUBY
+  end
+
+  it 'registers an offense when redundant line continuations for hash' do
+    expect_offense(<<~'RUBY')
+      {foo: \
+            ^ Redundant line continuation.
+        bar}
+    RUBY
+
+    expect_correction(<<~RUBY)
+      {foo:#{trailing_whitespace}
+        bar}
+    RUBY
+  end
+
+  it 'registers an offense when redundant line continuations for method call' do
+    expect_offense(<<~'RUBY')
+      foo(bar, \
+               ^ Redundant line continuation.
+        baz)
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo(bar,#{trailing_whitespace}
+        baz)
+    RUBY
+  end
+
+  it 'does not register an offense when line continuations inside comment' do
+    expect_no_offenses(<<~'RUBY')
+      # foo \
+      #  .bar
+    RUBY
+  end
+
+  it 'does not register an offense when string concatenation' do
+    expect_no_offenses(<<~'RUBY')
+      foo('bar' \
+        'baz')
+      foo(bar('string1' \
+          'string2')).baz
+    RUBY
+  end
+
+  it 'does not register an offense when line continuations are not redundant' do
+    expect_no_offenses(<<~'RUBY')
+      foo \
+        && bar
+      foo \
+        || bar
+      foo \
+        ? bar
+          : baz
+      1 \
+        + 2 \
+          - 3
+      some_method \
+        (argument)
+      some_method \
+        argument
+    RUBY
```

Comment:
Too many test cases may be written. Can you please split it for line continuations are not redundant?